api/src/main/java/io/grpc/InternalStatus.java

```diff
@@ -42,8 +42,8 @@ private InternalStatus() {}
    * of the stack trace.
    */
   @Internal
-  public static final StatusRuntimeException asRuntimeException(Status status,
```

Comment:
Should we tweak the name here to make it clear at call sites it is special? This PR is an improvement as-is, so it is up to you.

Reply:
Btw, a related question: is it important to keep the old signature deprecated? That's only for situations if some application has "api" and "core" from different versions, which is most likely unsupported but possible theoretically.

Reply:
I do consider it at times, especially with lesser-used artifacts like grpclb or xds. But it is pretty rare to depend on api and not core (transitively). grpc-context-only usages do happen, but I don't expect we'd be saving many people much headache. I suspect there are pretty frequent internal breakages between api and core that nobody considers/notices.

I see now we could have kept the old API and just chose which implementation based on `fillInStackTrace`. But it's also the sort of thing "why would anyone use this API except to disable the stack trace?"

Although... we don't actually need InternalStatusRuntimeException in api any more, because it is just using public API. This is good enough. Let's get it in to fix animalsniffer and anything else can be followup if we so desire.